src/main/java/com/google/sps/servlets/RecommendationServlet.java

```diff
@@ -35,7 +41,28 @@ public void init() {
   
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    // Get details for recommendation 
+    Long id = Long.parseLong(request.getParameter("id"));
+    
+    try {
+      Key key = KeyFactory.createKey("Recommendation", id);
+      Entity recommendation = datastore.get(key);
+      // Get details of recommendation
+      String name = (String) recommendation.getProperty("place-name");
+      String category = (String) recommendation.getProperty("category");
+      double lat = (double) recommendation.getProperty("latitude");
+      double lng = (double) recommendation.getProperty("longitude");
+      String description = (String) recommendation.getProperty("description");
+      int costRating = (int) (long) recommendation.getProperty("cost-rating");
+      int crowdRating = (int) (long) recommendation.getProperty("crowd-rating");
```

Comment:
Would be nice if you please clarify here why we are casting twice :) [Example](https://github.com/googleinterns/step241-2020/blob/master/src/main/java/com/google/sps/servlets/UserDataServlet.java#L50-L51)

Reply:
Good point, thank you!